hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -521,6 +521,14 @@ public boolean hasRowId() {
 		return propertySelectable;
 	}
 
+	public String[] getTableNames() {
+		String[] tableNames = new String[ getTableSpan() ];
```

Comment:
I discussed the opportunity of using a collector pattern (not sure it's the best solution, just mentioning it) but it's typically to avoid this.

Moreover, you do this work everytime you will call the method, and AFAICS the method will be called several times (e.g. for each query referencing this persister). That should be avoided if it can be.

Reply:
I do not know collector pattern, please suggest another implementation of the method.

Reply:
Usually it is preferred in the persisters to compute these values once, and then store them in a field. This because they are accessed many times.

For example: look at the method `doLateInit()` in `AbstractEntityPersister` which is invoked by `postInstantiate()`. I think this could be spot to instantiate the array as well.